src/compute/src/compute_state.rs

```diff
@@ -294,6 +308,34 @@ impl ComputeState {
         // Remember the maintenance interval locally to avoid reading it from the config set on
         // every server iteration.
         self.server_maintenance_interval = COMPUTE_SERVER_MAINTENANCE_INTERVAL.get(config);
+
+        let offset = COMPUTE_REPLICA_EXPIRATION_OFFSET.get(&self.worker_config);
```

Comment:
As we discussed, this is needed in tests. The compromise is to only allow setting a new expiration time when it is smaller than the existing one. This would be least surprising to anyone following the history of expiration times.